File: Data Structures & Algorithms/house-robber/submission-4.java
class Solution {
    private int[] memo;
    public int rob(int[] nums) {
        
        int n=nums.length;
        memo=new int[n];
        
        memo[0]=nums[0];
        if(n>1) memo[1]=Math.max(nums[0],nums[1]);

        for(int i=2;i<n;i++)
        {
            memo[i]=Math.max(memo[i-1],memo[i-2]+nums[i]);
        }
        return memo[n-1];

    }
    
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        res = 0
        nums.sort()
        for i in range(len(nums)):
                
            prev = nums[i]
            count = 1
            
            for j in range(len(nums)):

                if i == j:
                    continue
                if prev + 1 == nums[j]:
                    prev = nums[j]
                    count+=1
            res = max(count,res)

        return res


                



        
        